### PR TITLE
Update IPVS doc

### LINKED ISSUE #59505: IPVS needs better documentation
/kind documentation

**What happened**:
IPVS needs to be documented more.  Specifically, the existing documentation needs to be updated to properly describe how to bring up a cluster with ipvs in the kubeadm, local and gce case. Furthermore, there needs to be another piece of documentation that describes the main differences between iptables and ipvs and also dives into implementation differences between the two (ex. ipset). 

It would also be really nice if there could be some screenshots detailing the difference between iptables and ipvs rules (i.e bring up a cluster, create a service, dump iptables and ipvs rules, compare)

All this documentation should be added to pkg/proxy/ipvs

/assign @m1093782566 
/area ipvs
/sig network

### PR DESCRIPTION
**Pull request by islinwb**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #59505

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/proxy/ipvs/README.md [hunk 1]
```diff
@@ -65,26 +109,14 @@ UDP  10.0.0.10:53 rr
...
+Use the following check list to help you solve the problems: 
 
-**2. Specify proxy-mode=ipvs**
+**1. Specify proxy-mode=ipvs**
 
-Tell kube-proxy that proxy-mode=ipvs, please.
+Check whether the kube-proxy mode has been set to `ipvs`.
 
-**3. Load ipvs required kernel modules**
+If you are using Kubernetes v1.8, you should also enable IPVS feature gateway (set `--feature-gates=SupportIPVSProxyMode=true` explicitly).
```
Comment (m1093782566) [inline pkg/proxy/ipvs/README.md]: v1.8 -> v1.10 and below.

File: pkg/proxy/ipvs/README.md [hunk 2]
```diff
@@ -45,6 +77,18 @@ kubeProxy:
     mode: ipvs
 ...
 ```
+before running
+
+`kube init --config <path_to_kubeadm_config_file>`
+
+If you are using Kubernetes v1.8, you can also add the flag `--feature-gates=SupportIPVSProxyMode=true` in `kubeadm init` command
```
Comment (islinwb) [inline pkg/proxy/ipvs/README.md]: @m1093782566 You mean v1.8 --> v1.10 in this line too? For Kubernetes v1.9 and later, the flag `--feature-gates=SupportIPVSProxyMode=true` has been removed in `kubeadm init` .

Comment (m1093782566) [reply pkg/proxy/ipvs/README.md]: Okay, could you please mentioned it in somewhere?

Comment (islinwb) [reply pkg/proxy/ipvs/README.md]: :ok:

**Comment by rramkumar1** [comment]
@islinwb It would be nice if you would also be able to add a section which shows the ipvs rules for a simple clusterIP service such as kube-dns. Maybe even show the iptables rules that are also created for this service. This would be helpful debugging information

Furthermore, can you elaborate in the documentation on what situations ipvs falls back to iptables rules? Again, this would be helpful debugging info

**Comment by islinwb** [comment]
@rramkumar1 I'll try :)

File: pkg/proxy/ipvs/README.md [hunk 3]
```diff
@@ -10,30 +10,220 @@ Linux kernel.
 IPVS runs on a host and acts as a load balancer in front of a cluster of real servers. IPVS can direct requests for TCP
 and UDP-based services to the real servers, and make services of real servers appear as virtual services on a single IP address.
 
+## IPVS vs. IPTABLES
+IPVS mode was introduced in Kubernetes v1.8 and goes beta in v1.9. IPTABLES mode was added in v1.1 and become the default operating mode since v1.2. Both IPVS and IPTABLES are based on `netfilter`.
+Differences between IPVS mode and IPTABLES mode are mainly as follows:
```
Comment (rramkumar1) [inline pkg/proxy/ipvs/README.md]: Nit: Remove "mainly"

File: pkg/proxy/ipvs/README.md [hunk 4]
```diff
@@ -10,30 +10,220 @@ Linux kernel.
...
+3. IPVS supports more sophisticated load balancing algorithms than iptables (least load, least connections, locality, weighted, etc.).  
+
+4. IPVS supports server health checking and connection retries, etc.
+ 
+### When ipvs falls back to iptables
+IPVS proxier will employ iptables in doing packet filtering, SNAT and supporting NodePort type service. Specifically, ipvs proxier will fall back on iptables in the following 4 scenarios.
+
+**1. kube-proxy start with --masquerade-all=true**
+
+If kube-proxy start with `--masquerade-all=true`, ipvs proxier will masquerade all traffic accessing service Cluster IP, which behaves same as what iptables proxier does. Suppose there is a serivice with Cluster IP `10.244.5.1` and port `8080`, then the iptables installed by ipvs proxier should be like what is shown below.
```
Comment (rramkumar1) [inline pkg/proxy/ipvs/README.md]: Nit: "start" -> "starts"

Comment (rramkumar1) [inline pkg/proxy/ipvs/README.md]: Nit: "behaves the same as what iptables proxier does" -> "behaves the sames as the iptables proxier"

Comment (rramkumar1) [inline pkg/proxy/ipvs/README.md]: Nit "serivice" -> "service"

File: pkg/proxy/ipvs/README.md [hunk 5]
```diff
@@ -10,30 +10,220 @@ Linux kernel.
...
+MARK       all  --  0.0.0.0/0            0.0.0.0/0            MARK or 0x4000
+
+Chain KUBE-SERVICES (2 references)
+target     prot opt source               destination         
+KUBE-MARK-MASQ  tcp  -- 0.0.0.0/0        10.244.5.1            /* default/foo:http cluster IP */ tcp dpt:8080
+```
+
+**2. Specify cluster CIDR in kube-proxy startup**
+
+If kube-proxy start with `--cluster-cidr=<cidr>`, ipvs proxier will masquerade off-cluster traffic accessing service Cluster IP, which behaves same as what iptables proxier does. Suppose kube-proxy is provided with the cluster cidr `10.244.16.0/24`, and service Cluster IP is `10.244.5.1` and port is `8080`, then the iptables installed by ipvs proxier should be like what is shown below.
```
Comment (rramkumar1) [inline pkg/proxy/ipvs/README.md]: Nit: "start" -> "starts"

Comment (rramkumar1) [inline pkg/proxy/ipvs/README.md]: Nit: "behaves the same as what iptables proxier does" -> "behaves the sames as the iptables proxier"

File: pkg/proxy/ipvs/README.md [hunk 6]
```diff
@@ -10,30 +10,220 @@ Linux kernel.
...
+KUBE-SEP-Q3UCPZ54E6Q2R4UT  all  --  0.0.0.0/0            0.0.0.0/0            /* default/nginx-service:http */
+Chain KUBE-SEP-Q3UCPZ54E6Q2R4UT (1 references)
+target     prot opt source               destination         
+KUBE-MARK-MASQ  all  --  172.17.0.2           0.0.0.0/0            /* default/nginx-service:http */
+DNAT       tcp  --  0.0.0.0/0            0.0.0.0/0            /* default/nginx-service:http */ tcp to:172.17.0.2:80
+```
 ## Run kube-proxy in ipvs mode
 
-Currently, local-up scripts and kubeadm support switching IPVS proxy mode via exporting environment variables or specifying flags.
+Currently, local-up scripts and kubeadm support switching IPVS proxy mode via exporting environment variables or specifying flags.
```
Comment (rramkumar1) [inline pkg/proxy/ipvs/README.md]: Gce scripts also now support it

File: pkg/proxy/ipvs/README.md [hunk 7]
```diff
@@ -10,30 +10,220 @@ Linux kernel.
...
+ip_vs_wrr
+ip_vs_sh
+nf_conntrack_ipv4
+```
+Packages such as `ipset` should also be installed on the node before using IPVS mode.  
 
-Users should export the env `KUBE_PROXY_MODE=ipvs` to specify the ipvs mode before deploying the cluster if want to run kube-proxy in ipvs mode.
+Kube-proxy will fall back to IPTABLES mode if those requirements are not met.
 
-### Cluster Created by Kubeadm
```
Comment (rramkumar1) [inline pkg/proxy/ipvs/README.md]: We should add a cluster created by GCE as well since we have plumbed the envvar through in those scripts/

**Review by rramkumar1 — COMMENTED** [review]
First pass, looks a lot better so far!

**Comment by islinwb** [comment]
@rramkumar1 PTAL

**Comment by m1093782566** [comment]
/approve

Leave @rramkumar1 for lgtm :)

**Comment by m1093782566** [comment]
testing....

**Comment by m1093782566** [comment]
testing milestone again……
/milestone v1.11

**Comment by m1093782566** [comment]
great，milestone command is fixed now！

**Comment by rramkumar1** [comment]
@jdumars How do we get milestone/incomplete-labels removed?

**Pull request merged**